### PR TITLE
refactor: add segment tracking for git sync

### PR DESCRIPTION
**Pull request by marckong**
- adding a oauth provider name type
- adding a util function to return oauth provider naem
- modifying the segments for git flow to add provider name

for github,
image

for gitlab,
image

### CODE REVIEW

File: packages/insomnia/src/sync/git/utils.ts [hunk 1]
```diff
@@ -95,6 +96,14 @@ const onAuth = (credentials?: GitCredentials): AuthCallback => (): GitAuth => {
   };
 };
 
+export const getOauth2FormatName = (credentials?: GitCredentials | null): OauthProviderName | undefined => {
+  if (credentials && 'oauth2format' in credentials) {
+    return credentials.oauth2format;
+  }
+
+  return undefined;
```
Comment (marckong) [inline packages/insomnia/src/sync/git/utils.ts]: Or we could return null but if we return null then we have to null check in the consumer functions.

Comment (johnwchadwick) [reply packages/insomnia/src/sync/git/utils.ts]: If you want it to look a bit more ergonomic, a `return;` also returns undefined.

Comment (marckong) [reply packages/insomnia/src/sync/git/utils.ts]: Good point! was initially returning null and just dumb-replaced null with undefined during a zoom call lol

**Review by wdawson — APPROVED** [review]
Looks good to me, thanks @marckong ! Did we also confirm that these events are firing in all cases where a user can do a git clone (e.g. from the dashboard, from an existing document, etc.)?

**Review by filfreire — APPROVED** [review]

**Review by filfreire — APPROVED** [review]
Aproved. `type-checks` are failing but I think we can merge once you fix that failure.

**Pull request merged**